File: code/c/types/test-enum.c
/* Program: test-enum.c */

typedef enum
{
    SAFE,
    DANGER,
    EXTREME_DANGER
} warning_level;

int main()
{
    warning_level situation = SAFE;
    
    switch (situation)
    {
        case SAFE: printf("Safe\n"); break;
        case DANGER: printf("Danger!\n"); break;
        case EXTREME_DANGER: printf("Run!\n"); break;
        default: printf("Unknown...\n");
    }
    return 0;
}

File: code/pascal/types/TestEnum.pas
program TestEnum;

type
    WarningLevel = ( SAFE, DANGER, EXTREME_DANGER );

procedure Main();
var
    situation: WarningLevel = SAFE;
begin
    case situation of
        SAFE: WriteLn('Safe');
        DANGER: WriteLn('Danger!');
        EXTREME_DANGER: WriteLn('Run!');
        else WriteLn('Unknown...');
    end;
end;

begin
    Main();
end.